### ISSUE TITLE
Add autocompletion to search box

### ISSUE THREAD

**Issue by literarymachine** [hbz/oerworldmap#1125]

**Comment by karindr** [comment]
This would be really helpful because at the moment we have no (or nearly no) truncation so when I search for example for "iright" I get no entry as result only "irights" gets me to iRights e.V.

**Comment by trugwaldsaenger** [comment]
+1 I still think the search-functionality needs improvement. What do you think @acka47? You recently mentioned, that the inclusion of a "Did you mean..." functionality would be helpful as well. Would we still need this, if we had autocompletion? Maybe both extenions could be combined, so that even users, which ignore the autocompletion get good results...

**Comment by acka47** [comment]
Yes, I think both functions do make sense combined. People know it like this from google.Re. the "Did you mean..." feature: It could be a good solution to some recurring discussions we have on search:, on the one side wanting to have some fuzzy search features while on the other side the result list should rather be short. A "Did you mean..." feature would give users benefit from fuzzy search without letting the result list get to big. I will open a separate issue for this.

**Comment by trugwaldsaenger** [comment]
+1 Many thanks Adrian! @philboeselager : Do you think this will be difficult to implement?

**Comment by philboeselager** [comment]
> @philboeselager : Do you think this will be difficult to implement?

No, not really. Let's talk about the best way to implement it, @literarymachine

**Comment by literarymachine** [comment]
> No, not really. Let's talk about the best way to implement it, @literarymachine

Auto completion (and auto-correction, i.e. "Did you mean ...") are non-trivial if we want a Google-like multi-word approach *where the words don't come from a phrase in a single field*. I implemented something like this in another project, see the search box in http://primo.bibliothek.kit.edu/primo_library/libweb/action/search.do?vid=KIT&tab=kit&srt=rank. That approach is based on aggregations. I'm happy to accept other solutions though!
